lib/templates.php

```diff
@@ -183,7 +183,7 @@ function filter_rest_wp_template_collection_params( $query_params ) {
 function filter_rest_wp_template_query( $args, $request ) {
 	if ( $request['resolved'] ) {
 		$template_ids   = array( 0 ); // Return nothing by default (the 0 is needed for `post__in`).
-		$template_types = $request['slug'] ? array( $request['slug'] ) : get_template_types();
+		$template_types = $request['slug'] ? $request['slug'] : get_template_types();
```

Comment:
This is the bug I missed in #21981. Turns out that `$request['some-param']` is an array :grimacing: